### PR TITLE
New Bidder: Adf

### PR DESCRIPTION
**Pull request by nickluck8**

### CODE REVIEW

**Comment by muuki88** [comment]
@nickluck8 there's already an [AdForm bidder](https://github.com/prebid/prebid-server-java/blob/master/src/main/java/org/prebid/server/bidder/adform/AdformBidder.java) available.

Can you highlight the difference and which one should be chosen? From the implementation I can see that
the `adf` one uses the standard `OpenrtbBidder`, which is great. So in general it would be nice to know 

- can both adapters be used?
- is there a difference in formats, demand, future support?

**Comment by rpanchyk** [comment]
@muuki88 seems `ADF` - `https://adx.adform.net/adx/openrtb` is OpenRTB exchange instead of `AdForm` - `https://adx.adform.net/adx` where custom GET request used.
So, it depends on publisher which one to use.

**Review by snahornyi — CHANGES_REQUESTED** [review]
Seems this PR should be updated, with this changes https://github.com/prebid/prebid-server/pull/1841

File: src/main/java/org/prebid/server/bidder/adf/AdfBidder.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.iab.openrtb.response.Bid;
+import org.prebid.server.bidder.OpenrtbBidder;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.request.adf.ExtImpAdf;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+
+import java.util.List;
+
+public class AdfBidder extends OpenrtbBidder<ExtImpAdf> {
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/adf/AdfBidder.java:13]: Minor. 
```java
/**
 * Adf {@link Bidder} implementation.
 */
```

File: src/test/resources/org/prebid/server/it/openrtb2/adf/test-auction-adf-request.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  },
+  "at": 1,
+  "tmax": 5000,
+  "cur": [
+    "USD"
+  ],
+  "source": {
+    "fd": 1,
+    "tid": "tid"
+  },
```
Comment (snahornyi) [inline src/test/resources/org/prebid/server/it/openrtb2/adf/test-auction-adf-request.json:41]: Can be removed

File: src/test/resources/org/prebid/server/it/openrtb2/adf/test-adf-bid-request.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+      "id": "impId001",
+      "native": {
+        "request": "{\"ver\":\"1.2\",\"context\":1,\"plcmttype\":4,\"plcmtcnt\":1,\"assets\":[{\"id\":2,\"required\":1,\"title\":{\"len\":90}},{\"id\":6,\"required\":1,\"img\":{\"type\":3,\"wmin\":128,\"hmin\":128,\"mimes\":[\"image/jpg\",\"image/jpeg\",\"image/png\"]}},{\"id\":7,\"required\":1,\"data\":{\"type\":2,\"len\":120}}]}",
+        "ver": "1.2"
+      },
+      "tagid": "1",
+      "ext": {
+        "bidder": {
+          "mid": "1"
```
Comment (snahornyi) [inline src/test/resources/org/prebid/server/it/openrtb2/adf/test-adf-bid-request.json:13]: Let's remove tagId in auction request, and check that masterId is being added as tagid

**Review by snahornyi — CHANGES_REQUESTED** [review]

**Review by snahornyi — APPROVED** [review]

**Pull request merged**